File: Easy/1217_Minimum_Cost_to_Move_Chips_to_The_Same_Position/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    int minCostToMoveChips(vector<int>& position) {
        int even_count = 0;
        int odd_count = 0;

        for (int i : position) {
            if (i % 2 == 0) {
                even_count++;
            } else {
                odd_count++;
            }
        }

        return min(even_count, odd_count);
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    vector<int> position1 = {1, 2, 3};
    cout << solution.minCostToMoveChips(position1) << endl;

    // test cases 2
    vector<int> position2 = {2, 2, 2, 3, 3};
    cout << solution.minCostToMoveChips(position2) << endl;

    // test cases 3
    vector<int> position3 = {1, 1000000000};
    cout << solution.minCostToMoveChips(position3) << endl;

    return 0;
}

File: Easy/1217_Minimum_Cost_to_Move_Chips_to_The_Same_Position/Solution.java


public class Solution {
    public int minCostToMoveChips(int[] position) {
        int evenCount = 0;
        int oddCount = 0;

        for (int chip : position) {
            if (chip % 2 == 0) {
                evenCount++;
            } else {
                oddCount++;
            }
        }

        return Math.min(evenCount, oddCount);
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        int[] position1 = {1, 2, 3};
        System.out.println(solution.minCostToMoveChips(position1)); // Expected output: 1
        
        // test cases 2
        int[] position2 = {2, 2, 2, 3, 3};
        System.out.println(solution.minCostToMoveChips(position2)); // Expected output: 2

        // test cases 3
        int[] position3 = {1, 1000000000};
        System.out.println(solution.minCostToMoveChips(position3)); // Expected output: 1
        
    }
}